docs/architecture.rst

```diff
@@ -1,142 +1,211 @@
-  secrets from a variety of providers (e.g. AWS, Azure, GCP, Hugging Face, Github, etc.) as well as SSH Keys and
-  custom secrets, and stores them in Hashicorp Vault.
+    # [6]
+    gpu.down()
 
-* **Configs**: Set and preserve default configs across environments.
+  .. note::
 
-Management
-----------
 
-Runhouse provides tools for visibility and management of resources as long-living assets shared by teams or projects.
+1. **Specify and/or Allocate Compute**: Runhouse can allocate compute to the application on the fly, either by
+    utilizing an existing VM or Ray cluster, or allocating one fresh using local cloud or K8s credentials. The
+    `rh.cluster` constructor is generally used to specify and interact with remote compute, including bringing it up
+    if necessary (`cluster.up_if_not()`).
+2. **Starting the Runhouse Server Daemon**: If not already running, the client will start the Runhouse API server daemon
+    on the compute and form a network connection (either over SSH or HTTP/S). Dependencies can be specified to be
+    installed before starting the daemon.
+        a. The daemon can be thought of as a "Python object server", holding key-value pairs of names and Python
+            objects in memory, and exposing an HTTP API to call methods on those objects by name.
+        b. The objects are held in a single default worker process by default, but can be sent to other worker
+            processes, including on other nodes in the cluster, to achieve powerful parallelism out of the box.
+        c. If I call GET http://myserver:32300/my_object/my_method, the daemon will look up the object named
+            "my_object", issue an instruction for its worker to call the method "my_method" on it, and
+            return the result.
+        d. The HTTP server and workers can handle thousands of concurrent calls per second, and have similar latency
+            under simple conditions to Flask.
+        e. New workers can be constructed with `rh.env`, which specifies the details of the Python environment
+            (packages, environment variables) in which the process will be constructed. By default workers live
+            in the same Python environment as the daemon, but can also be started in a conda environment or a
+            separate node.
+3. **Deploying Functions or Classes**: The user specifies a function or class to be deployed to the remote compute
+    using the `rh.function` or `rh.module` constructors (or by subclassing `rh.Module`), and calling
+    `remote_obj = my_obj.to(my_cluster, worker=my_env)`. The Runhouse client library extracts the path, module name,
+    and importable name from the function or class. If the function or class is defined in local code, the repo or
+    package is rsynced onto the cluster. An instruction with the import path is sent to the cluster to
+    construct the function or class in a particular worker and assign it a name in the key-value store, perhaps
```

Comment:
"upserts it into the key-value store." might be more concise?

Reply:
+ 1 feels more like an implementation detail - at a higher level this also shows how easy it is to update and redeploy an existing service